### PR TITLE
Streamlined target platform for RSP

### PR DESCRIPTION
**Pull request by robstryker**
Signed-off-by: Rob Stryker <rob@oxbeef.net>

### CODE REVIEW

File: framework/tests/org.jboss.tools.rsp.server.test/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -34,4 +34,4 @@ Require-Bundle: org.jboss.tools.rsp.api;bundle-version="0.0.9",
  org.jboss.tools.rsp.server.spi;bundle-version="0.0.9",
  org.jboss.tools.rsp.server;bundle-version="0.0.9",
```
Comment (adietish) [inline framework/tests/org.jboss.tools.rsp.server.test/META-INF/MANIFEST.MF]: shouldn't we require 0.10.0 bundles?

File: runtimes/tests/org.jboss.tools.rsp.server.wildfly.test/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -37,4 +37,4 @@ Import-Package: org.eclipse.lsp4j.jsonrpc,
 Require-Bundle: org.jboss.tools.rsp.server;bundle-version="0.0.9",
  org.jboss.tools.rsp.server.wildfly;bundle-version="0.0.9",
```
Comment (adietish) [inline runtimes/tests/org.jboss.tools.rsp.server.wildfly.test/META-INF/MANIFEST.MF]: shouldn't we require 0.10.0 bundles?

**Comment by adietish** [comment]
other than the complaints the TP works for me.

File: targetplatform/rsp-target.target [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      <unit id="org.eclipse.help.feature.group" version="2.2.300.v20180906-0745"/>
+      <unit id="org.eclipse.jdt.feature.group" version="3.15.0.v20180906-0745"/>
+      <unit id="org.eclipse.pde.feature.group" version="3.13.200.v20180906-0745"/>
+      <unit id="org.eclipse.platform.feature.group" version="4.9.0.v20180906-1121"/>
+      <unit id="org.eclipse.rcp.feature.group" version="4.9.0.v20180906-1121"/>
+
+      <!-- required for swtbot and reddeer -->
+      <!-- org.eclipse.linuxtools.docker.reddeer 2.1.0.201805161441 requires org.hamcrest.library 1.3.0 -->
+      <unit id="org.hamcrest.core" version="1.3.0.v20180420-1519"/>
+      <unit id="org.hamcrest.library" version="1.3.0.v20180524-2246"/>
```
[the remark this answers is not in the document]
Comment (robstryker) [reply targetplatform/rsp-target.target]: mockito uses hamcrest

[the remark this answers is not in the document]
Comment (adietish) [reply targetplatform/rsp-target.target]: ok, verified it:
https://github.com/mockito/mockito/blob/release/2.x/build.gradle#L77

So we should change the comment accordingly.

**Pull request merged**